kb/deadline-exceeded-troubleshooting.md

```diff
@@ -0,0 +1,117 @@
+---
+slug: deadline-exceeded-troubleshooting
+title: DeadlineExceeded error troubleshooting
+tags:
+  - kb-article
+date: 2022-11-09T00:00:00Z
+---
+
+All client-side requests made to the [Temporal Cluster](/concepts/what-is-a-temporal-cluster) are [gRPC requests](https://grpc.io/docs/what-is-grpc/core-concepts/#deadlines).
+Sometimes, when these frontend requests can't be completed, you'll see this particular error message: `Context: deadline exceeded`.
+Network interruptions, timeouts, server overload, and Query errors are some of the causes of this error.
+
+The following sections discuss the nature of this error and how to troubleshoot it.
+
+<!-- truncate -->
+
+#### Check Frontend Service logs
+
+:::note
+
+Cloud users cannot access some of the logs needed to diagnose the source of the error.
+
+If you're using Temporal Cloud, check your Workflow History for error messages and create a [support ticket](https://support.temporal.io/) with Temporal.
+Provide the full error message in your ticket.
+
+:::
+
+[Frontend Service](/clusters/#frontend-service) logs can show which parts of the Cluster aren't working.
+For the error to appear, a service pod or container must be up and running.
+
+OSS users can verify that the Frontend Service is connected and running by using `tctl`.
+
+```
+
+tctl --address frontendAddress:frontendPort cluster health
+
+```
+
+Use `grpc-health-probe` to check the Frontend Service, [Matching Service](/clusters#matching-service), and [History Service](/clusters#history-service).
+
+```
+
+./grpc-health-probe -addr=frontendAddress:frontendPort -service=temporal.api.workflowservice.v1.WorkflowService
+
+./grpc-health-probe -addr=matchingAddress:matchingPort -service=temporal.api.workflowservice.v1.MatchingService
+
+./grpc-health-probe -addr=historyAddress:historyPort -service=temporal.api.workflowservice.v1.HistoryService
+
+```
+
+Logs can also be used to find failed Client [Query](/workflows#queries) requests.
+
+#### Check your Cluster metrics
+
+Cluster metrics can be used to detect issues (such as `resource exhausted`) that impact Cluster health.
+A `resource exhausted` error can cause your client request to fail, which prompts the `deadline exceeded` error.
+
+Use the following query to check for errors in `RpsLimit`, `ConcurrentLimit` and `SystemOverloaded` on your metrics dashboard.
+
+```
+
+sum(rate(service_errors_resource_exhausted{}[1m])) by (resource_exhausted_cause)
```

Comment:
Probably worth explaining what this query language and that this requires prometheus metrics output.